engine/memng/mem.go

```diff
@@ -13,10 +13,11 @@ import (
 
 // Mem is exported to provide easy access to its internals
 type Mem struct {
-	Hosts     map[engine.HostKey]engine.Host
-	Frontends map[engine.FrontendKey]engine.Frontend
-	Backends  map[engine.BackendKey]engine.Backend
-	Listeners map[engine.ListenerKey]engine.Listener
+	Hosts             map[engine.HostKey]engine.Host
+	Frontends         map[engine.FrontendKey]engine.Frontend
+	Backends          map[engine.BackendKey]engine.Backend
+	Listeners         map[engine.ListenerKey]engine.Listener
+	SharedMiddlewares []engine.Middleware
```

Comment:
Ah good catch. You of course know where this came from. :-) It's a bad merge from my changes to do shared middlewares.